src/main/java/Nimbus/CommandParser.java

```diff
@@ -0,0 +1,97 @@
+package Nimbus;
+
+public class CommandParser {
+    public static String[] validCommands = {"bye", "list", "mark", "unmark", "todo", "deadline", "event", "-help", "delete", "save"};
+    private String commandWord;
+    private String arguments;
+
+    public CommandParser(String input, int range) throws NimbusException {
+        this.commandWord = null;
+        this.arguments = null;
+        String[] commands = input.trim().split(" ", 2);
+        if (commands.length == 0) {
+            throw new NimbusException("empty command");
+        }
+
+        this.commandWord = commands[0].toLowerCase();
+        if (!this.isValidCommandWord()) {
+            throw new NimbusException("invalid command word");
+        }
+        if (commands.length == 2) {
+            this.arguments = commands[1];
+        }
+
+        switch (this.commandWord) {
+        case "bye":
+        case "-help":
+        case "save":
+        case "list": {
+            if (commands.length != 1) {
+                throw new NimbusException("invalid usage");
+            }
+            break;
+        }
```

Comment:
You do not need to use braces to scope a case statement. The case statement format should just be:
```java
// from java coding standard website
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```